### ISSUE TITLE
It's not possible to run the `ls` command inside directories that contain characters "[]"

### ISSUE THREAD

**Issue by Wint3rmute** [nushell/nushell#9232]
### Describe the bug

When attempting to list files inside a directory which name contain characters `[` and `]`, nushell fails to display the file list.

### How to reproduce

I'm building the `main` branch at 18th May, but the behaviour is also reproducible on the Arch Linux `nushell` package (version `0.80.0`):

```bash
cargo r
mkdir "[test]"
cd "[test]"
ls # this will work as the folder is empty
touch file
ls # this will fail
Error: 
  × No matches found for *
   ╭─[entry #5:1:1]
 1 │ ls
   · ─┬
   ·  ╰── Pattern, file or folder not found
   ╰────
  help: no matches found
```

### Expected behavior

`nushell` should list the files present in the folder, no matter what characters are present in the folder's name.

### Screenshots

![image](https://github.com/nushell/nushell/assets/19196352/bd93f0cb-fb55-489f-bfaa-fa77883ecf30)

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.80.1                                   |
| branch             | main                                     |
| commit_hash        | e752d8a964595b28ef3cce5884cdbbab6cc62007 |
| build_os           | linux-x86_64                             |
| build_target       | x86_64-unknown-linux-gnu                 |
| rust_version       | rustc 1.67.1 (d5a82bbd2 2023-02-07)      |
| rust_channel       | 1.67.1-x86_64-unknown-linux-gnu          |
| cargo_version      | cargo 1.67.1 (8ecd4f20a 2023-01-10)      |
| build_time         | 2023-05-18 22:20:13 +02:00               |
| build_rust_channel | debug                                    |
| features           | default, zip                             |
| installed_plugins  |                                          |

**Comment by fdncred** [comment]
I probably broke this when I [updated ](https://github.com/nushell/nushell/pull/9099)the nu-glob crate a while back. I'm guessing nushell is getting confused because you have a directory named with glob syntax like `/some/path/[test]` so when you try to do a ls, it's like you're doing `ls /some/path/[test]`. I'm not sure what the solution should be but I'd hate to remove `[]` from nu-glob just so people can have directories named with brackets.

**Comment by bobhy** [comment]
Candidate, most entertaining bug of the month!
As a workaround, you *can* get a list of files in this directory via Nu `glob` itself!
```
..v| -------------------------/home/bobhy/work/w1/[test] -------------------------
〉glob *
╭───┬────────────────────────────────╮
│ 0 │ /home/bobhy/work/w1/[test]/foo │
╰───┴────────────────────────────────╯
```

**Comment by fdncred** [comment]
now that i think about it. this may be why we vendored rust's glob crate way back when. but unfortunately it limits how we can use globs. it would be nice to have a fix for this so we can use brackets in folder paths and still allow our ls to work. maybe there's some way to escape brackets in the code?

**Comment by fdncred** [comment]
If you're interested in a fix for this, could you please test #9416 and let me know how it goes?
